crates/ibc/src/core/ics02_client/error.rs

```diff
@@ -1,280 +1,140 @@
-                    e.client_type)
-            },
-
-        InsufficientVotingPower
-            { reason: String }
-            | e | {
-                format_args!("Insufficient overlap {}", e.reason)
-            },
-
-        RawClientAndConsensusStateTypesMismatch
-            {
-                state_type: ClientType,
-                consensus_type: ClientType,
-            }
-            | e | {
-                format_args!("mismatch in raw client consensus state {} with expected state {}",
-                    e.state_type, e.consensus_type)
-            },
-
-        LowHeaderHeight
-            {
-                header_height: Height,
-                latest_height: Height
-            }
-            | e | {
-                format!("received header height ({}) is lower than (or equal to) client latest height ({})",
-                    e.header_height, e.latest_height)
-            },
-
-        LowUpgradeHeight
-            {
-                upgraded_height: Height,
-                client_height: Height,
-            }
-            | e | {
-                format_args!("upgraded client height {} must be at greater than current client height {}",
-                    e.upgraded_height, e.client_height)
-            },
-
-        InvalidConsensusStateTimestamp
-            {
-                time1: Timestamp,
-                time2: Timestamp,
-            }
-            | e | {
-                format_args!("timestamp is invalid or missing, timestamp={0},  now={1}", e.time1, e.time2)
-            },
-
-        HeaderNotWithinTrustPeriod
-            {
-                latest_time:Timestamp,
-                update_time: Timestamp,
-            }
-            | e | {
-                format_args!("header not withing trusting period: expires_at={0} now={1}", e.latest_time, e.update_time)
-            },
-
-        MissingLocalConsensusState
-            { height: Height }
-            | e | { format_args!("the local consensus state could not be retrieved for height {}", e.height) },
-
-        InvalidConnectionEnd
-            [ TraceError<TendermintProtoError>]
-            | _ | { "invalid connection end" },
-
-        InvalidChannelEnd
-            [ TraceError<TendermintProtoError>]
-            | _ | { "invalid channel end" },
-
-        InvalidAnyClientState
-            [ TraceError<TendermintProtoError>]
-            | _ | { "invalid any client state" },
-
-        InvalidAnyConsensusState
-            [ TraceError<TendermintProtoError> ]
-            | _ | { "invalid any client consensus state" },
-
-        Signer
-            [ SignerError ]
-            | _ | { "failed to parse signer" },
-
-        Ics23Verification
-            [ Ics23Error ]
-            | _ | { "ics23 verification failure" },
-
-        ClientSpecific
-            { description: String }
-            | e | { format_args!("client specific error: {0}", e.description) },
+#[derive(Debug, Display)]
+pub enum ClientError {
```

Comment:
Here are the error variants that we still have that are unused:
+ `InvalidRawConsensusState`
+ `MissingHeight`

Reply:
done